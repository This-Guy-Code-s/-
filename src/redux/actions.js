import axios from 'axios'



export const LOADING = 'LOADING'
export const SUCCESS = 'SUCCESS'
export const BTN = 'BTN'

export const NAVIGATION = 'NAVIGATION'


export const MODE = 'MODE'

export const SHOW = 'SHOW'

export const ERR = 'ERR'




//=============================================================================/

export const getMyWork = () => dispatch => {
//get my work that ill be displaying on my page
dispatch({type:LOADING})
axios.get(process.env.REACT_APP_mywork)
.then(res=>{
return dispatch({type:SUCCESS,payload:res.data})

})
.catch(err=>{
return dispatch({type:ERR,payload:err})

})
}


//=============================================================================/
//no dispatch throwing needed , but can be store here for good use of redux aysnc control
export const tog_description = (id) => dispatch =>{
const target = document.querySelector(`#${id}`);
//--------------------------------------------
(/off/i).test(target.getAttribute('class'))?
target.setAttribute('class','fas fa-toggle-on')
:target.setAttribute('class','fas fa-toggle-off');	
}

//=============================================================================/


export const changeBtnLabel = label => dispatch =>{

return dispatch({type:BTN,payload:label})
}

//=============================================================================/

export const navBar = whereTo => dispatch =>{
switch(whereTo){
case 'Home':
window.sessionStorage.setItem('nav',whereTo)
setTimeout(()=>{

	let view = '#intro'
	document.querySelector(view).scrollIntoView()
},500)
return dispatch({type:NAVIGATION,payload:whereTo})

case 'Projects':
window.sessionStorage.setItem('nav',whereTo)
setTimeout(()=>{

	let view = '#intro-proj'
	document.querySelector(view).scrollIntoView()
},500)
return dispatch({type:NAVIGATION,payload:whereTo})

default:
return false
}
}
//=============================================================================/

export const showOrNot = choice => dispatch =>{
return dispatch({type:SHOW,payload:!choice})
}

//=============================================================================/
export const DarkMode = mode => dispatch =>{
if(mode){
window.sessionStorage.setItem('toggleDarkMode',mode)
return dispatch({type:MODE,payload:mode})
}else{ 
window.sessionStorage.setItem('toggleDarkMode',mode)
return dispatch({type:MODE,payload:mode})
}  

}







 
  window.onload=()=>{
    document.addEventListener("contextmenu", (e)=>{
      e.preventDefault();
    }, false);
    document.addEventListener("keydown", (e)=>{
      if (e.ctrlKey && e.shiftKey && e.keyCode === 73) {
        disabledEvent(e);
      }
      if (e.ctrlKey && e.shiftKey && e.keyCode === 74) {
        disabledEvent(e);
      }
      if (e.keyCode === 83 && (navigator.platform.match("Mac") ? e.metaKey : e.ctrlKey)) {
        disabledEvent(e);
      }
      if (e.ctrlKey && e.keyCode === 85) {
        disabledEvent(e);
      }
      if (e.keyCode === 123) {
        disabledEvent(e);
      }
    }, false);
    const disabledEvent = (e) => {
      if (e.stopPropagation){
        e.stopPropagation();
      } else if (window.event){
        window.event.cancelBubble = true;
      }
      e.preventDefault();
      return false;
    }
  };