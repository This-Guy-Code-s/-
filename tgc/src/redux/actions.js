import axios from 'axios'

export const MOVING = 'MOVING'





export const getMyWork=()=>{
	axios.get('https://thisguycodez.herokuapp.com/api/myWork')
	.then(res=>{
		console.log(res)
	})
}