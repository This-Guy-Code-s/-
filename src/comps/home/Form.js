import React from 'react'
import axios from 'axios'
import {Pass,Err,xxx} from '../../util/contactStyles'
import {FormText,FormGroup} from 'reactstrap'
import {changeBtnLabel,showOrNot} from '../../redux/actions'
import {connect} from 'react-redux'
import './form.css'



class Form_ extends React.Component{
constructor(props){
super(props)
this.state={
valOrNah1:'',valOrNah2:'',valOrNah3:'',
msg:{name:'',email:'',message:''},
errMsg:'',passMsg:''
}

this.toInbox = this.toInbox.bind(this)
this.toggle = this.toggle.bind(this)

}





 toInbox(){


return axios.post(process.env.REACT_APP_sendmail
,this.state.msg)
.then(res=>{
console.log(res)
//reset entire form and state
 this.setState({passMsg:res.data})


 setTimeout(()=>{
 	return this.toggle()
 },3000)
}).catch(err=>{this.setState({errMsg:"server overwhelmed please wait or try again",passMsg:""})})


	}


  toggle(e){
if (this._isMounted){

this.setState({
valOrNah1:'',valOrNah2:'',valOrNah3:'',
msg:{name:'',email:'',message:''},
errMsg:'',passMsg:''
})


this.props.showOrNot(this.props.show)
   

this.props.show?this.props.changeBtnLabel('Hide..'):this.props.changeBtnLabel('Contact Me');

if(document.querySelector('#toggler'))
document.querySelector('#toggler').click()
 }
}




componentWillUnmount(){
	   this._isMounted = false;
}

render(){

return(


	    <div className="registration-form">
        <form  onSubmit={(e)=>{e.preventDefault();this.toInbox();return e.target.reset()}} data-testid="theForm">
            <div className="me-icon">
                <span><img src={this.props.me} width='100%' height='100%' alt='small me'/></span>
            </div>


            <FormGroup>
					<div className="form-input-group form-group">
						<div className="form-input-group-prepend">
							<span className="input-group-text" ><i className="fas fa-pen-square" ></i></span>
						</div>
						<input className={`form-control ${this.state.valOrNah1}`} type="text" name="name" id="name"  onChange={(e)=>{this.setState({msg:{...this.state.msg,name:e.target.value}})}} placeholder="Name..."  minLength={3} required/>
						
					</div>

					<div className="form-input-group form-group">
						<div className="form-input-group-prepend">
							<span className="input-group-text" ><i className="fas fa-pen-square" ></i></span>
						</div>
						<input className={`form-control ${this.state.valOrNah2}`}  type="email" name="email" id="email"  onChange={(e)=>{this.setState({msg:{...this.state.msg,email:e.target.value}})}} placeholder="Email..." minLength={8} required/>
						
					</div>


					<div className="form-input-group form-group">
						<div className="form-input-group-prepend">
							<span className="input-group-text" ><i className="fas fa-comment-dots" ></i></span>
						</div>
						<textarea className={`form-control ${this.state.valOrNah3}`} type="textarea" name="message" id="message"  onChange={(e)=>{this.setState({msg:{...this.state.msg,message:e.target.value}})}} placeholder="Message..." minLength={10} required></textarea>
					</div>
					
					<div className="form-group">
						<input type="submit" value="Send" className="btn btn-block send2me" />
					</div>
					<div className="card-footer">
				<div className="d-flex justify-content-center form-links">
					<FormText>
					<Err>{this.state.errMsg}</Err>
					<Pass>{this.state.passMsg}</Pass>
					</FormText>
				</div>
			</div>
					</FormGroup>


        </form>
        <div className="social-media">
            <h5>Other Options</h5>
        
				<div className="d-flex justify-content-end social-icons social_icon">
					<a href='tel:+12402737952' ><i className="fas fa-mobile-alt xxx" style={xxx}></i></a>
					<a href='mailto:thisguycodez@gmail.com' ><i className="fab fa-google-plus-square xxx" style={xxx}></i></a>
					<a href='https://www.linkedin.com/in/guytonoriji' ><i className="fab fa-linkedin xxx" style={xxx}></i></a>
				</div>

        </div>
    </div>
)
}
}


const mapStateToProps = state =>{
return {
...state
}
}

export default connect(
mapStateToProps,
{changeBtnLabel,showOrNot}
)(Form_)
