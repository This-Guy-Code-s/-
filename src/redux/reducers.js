import {LOADING,SUCCESS,BTN,NAVIGATION,MODE,SHOW,ERR} from './actions'
 

const initialState = {
	compRendered:localStorage.getItem('nav') || 'Home',
	work:'',
	error:'',
	mode:localStorage.getItem('toggleDarkMode') && localStorage.getItem('toggleDarkMode') === 'true'? true:false,
	show:false,
	//contact btn name toggle
	btnL:'Contact Me'
}



export const reducer =  (state = initialState, action) =>{
		switch(action.type){
			case LOADING:
			return {
				...state,
			}

			case SUCCESS:
			return {
				...state,
				work:action.payload,
			}

			case BTN:
			return {
				...state,
				btnL:action.payload,
			}

			case NAVIGATION:
			return {
				...state,
				compRendered:action.payload,
			}

			case MODE:
			return {
				...state,
				mode:action.payload,
			}



			case SHOW:
			return {
				...state,
				show:action.payload,
			}
			

			case ERR:
			return {
				...state,
				error:action.payload,
			}

			default:
			return state
			
		}







	}