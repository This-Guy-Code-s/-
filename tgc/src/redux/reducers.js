import {MOVING,MOVED,ERR} from './actions'



const initialState = {
	work:'',
	error:''
}



export const reducer =  (state = initialState, action) =>{
		switch(action.type){
			case MOVING:
			return {
				...state,
			}

			case MOVED:
			return {
				...state,
				work:action.payload,
			}

			case ERR:
			return {
				...state,
				error:action.payload,
			}

			default:
			return state
			
		}







	}