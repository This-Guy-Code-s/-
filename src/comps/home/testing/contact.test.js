import React from 'react'
import ReactDOM from 'react-dom'
import ContactBtn from '../Contact'
import Form from '../Form'
import { render } from '@testing-library/react'
 
import {createStore,applyMiddleware} from 'redux'
import thunk from 'redux-thunk'
import {reducer} from '../../../redux/reducers'
import {Provider} from 'react-redux'


const store = createStore(reducer,applyMiddleware(thunk))


const props = {
	name:'Guy',
	email:'GuyTheGreat@gmail.com',
	message:'this is the message',
	
}






test('Form renders with out crashing', async ()=>{
	const {screen,getByTestId,container} = render(<Provider store={store}><Form {...props}/></Provider>)
			
		const theForm = getByTestId('theForm')


		expect(theForm).toBeTruthy()//renders
		expect(theForm).toMatchSnapshot()//matches snap shot of its self(code)
		expect(container.innerHTML).toMatch(/<form/)//form tag is converted version(HTML5) and not react

})		






test('labels render and have proper text titles', async ()=>{
	const {screen, fireEvent,getAllByPlaceholderText,getByText,getByLabelText} = render(<Provider store={store}><Form {...props}/></Provider>)
				
				let nameLabel = getByLabelText('Name:')//name label
				let emailLabel = getByLabelText('Email:')//email label
				let msgLabel = getByLabelText('Message:')//message label

				expect(nameLabel).toMatchSnapshot()
				expect(nameLabel).toBeTruthy()

				expect(emailLabel).toMatchSnapshot()
				expect(emailLabel).toBeTruthy()

				expect(msgLabel).toMatchSnapshot()
				expect(msgLabel).toBeTruthy()

})		





test('inputs render and can take values', async ()=>{
	const {screen, fireEvent,getAllByPlaceholderText,getByText,getByLabelText} = render(<Provider store={store}><Form {...props}/></Provider>)
			
				let nameVal = getAllByPlaceholderText('Name...')//name input
				let emailVal = getAllByPlaceholderText('Email...')//email input
				let msgVal = getAllByPlaceholderText('Message...')//msg input

				expect(nameVal.value).toEqual(undefined)//start out with no values(empty inputs)
				expect(emailVal.value).toEqual(undefined)//start out with no values(empty inputs)
				expect(msgVal.value).toEqual(undefined)//start out with no values(empty inputs)



				//value sent to inputs from props obj
				nameVal.value = props.name//giving the value of "Guy"
				emailVal.value = props.email//giving the value of "GuyTheGreat@gmail.com"
				msgVal.value = props.message//giving the value of "this is the message"

				//input values held change
				expect(nameVal.value).toMatch('Guy')
				expect(emailVal.value).toMatch('GuyTheGreat@gmail.com')
				expect(emailVal.value).toMatch(/^(([^<>()[\]\\.,;:\s@"]+(\.[^<>()[\]\\.,;:\s@"]+)*)|(".+"))@((\[[0-9]{1,3}\.[0-9]{1,3}\.[0-9]{1,3}\.[0-9]{1,3}])|(([a-zA-Z\-0-9]+\.)+[a-zA-Z]{2,}))$/)
				expect(msgVal.value).toMatch('this is the message')



})		


test('inputs "name" attribute values connects/equals its proper label tag with its "for" attributes value', async ()=>{
	const {container,screen, fireEvent,getAllByPlaceholderText,getByText,getByLabelText} = render(<Provider store={store}><Form {...props}/></Provider>)
			

			//turns string form component into a string then into an Array split by the parent div of each label tag
			//then each label tag was saved to its own variable to run RegExp test confirm the "for" attributes value
			//in the label tag matches the same value as the name attribute value in the input tags
			//console.log(container.innerHTML.split('<div class="form-group">'))

			const label2Input_name = container.innerHTML.split('<div class="form-group">')[1]
			const label2Input_email = container.innerHTML.split('<div class="form-group">')[2]
			const label2Input_msg = container.innerHTML.split('<div class="form-group">')[3]

			expect(/for="name"/.test(label2Input_name) && 
						/name="name"/.test(label2Input_name)).toEqual(true)

			expect(/for="email"/.test(label2Input_email) && 
						/name="email"/.test(label2Input_email)).toEqual(true)
			
			expect(/for="message"/.test(label2Input_msg) && 
						/name="message"/.test(label2Input_msg)).toEqual(true)



})		
