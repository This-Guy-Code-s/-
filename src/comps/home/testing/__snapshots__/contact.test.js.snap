// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Form renders with out crashing 1`] = `
<form
  class=""
  data-testid="theForm"
  style="background-color: rgba(0, 0, 0, 0.4); color: rgb(255, 255, 255); padding: 0px 10px; width: 320px;"
>
  <div
    class="form-group"
  >
    <label
      class=""
      for="name"
      style="font-weight: bolder;"
    >
      Name:
    </label>
    <input
      class="form-control"
      id="name"
      name="name"
      placeholder="Name..."
      required=""
      type="text"
      value=""
    />
    <small
      class="form-text text-muted"
    >
      What name can I refer to you as?
    </small>
  </div>
  <div
    class="form-group"
  >
    <label
      class=""
      for="email"
      style="font-weight: bolder;"
    >
      Email:
    </label>
    <input
      class="form-control"
      id="email"
      name="email"
      placeholder="Email..."
      required=""
      type="email"
      value=""
    />
    <small
      class="form-text text-muted"
    >
      You will not recieve any spam or emails period.
    </small>
  </div>
  <div
    class="form-group"
  >
    <label
      class=""
      for="message"
      style="font-weight: bolder;"
    >
      Message:
    </label>
    <textarea
      class="form-control"
      id="message"
      name="message"
      placeholder="Message..."
      required=""
    />
    <small
      class="form-text text-muted"
    >
      <span
        class="sc-AxjAm bfepVn"
      />
      <span
        class="sc-AxirZ CaKOU"
      />
    </small>
  </div>
  <button
    class="btn btn-secondary"
    style="background-color: rgba(0, 0, 0, 0.2); border: double azure; font-size: 1.3rem; cursor: pointer; margin-bottom: 15px;"
  >
    Send
  </button>
</form>
`;

exports[`labels render and have proper text titles 1`] = `
<input
  class="form-control"
  id="name"
  name="name"
  placeholder="Name..."
  required=""
  type="text"
  value=""
/>
`;

exports[`labels render and have proper text titles 2`] = `
<input
  class="form-control"
  id="email"
  name="email"
  placeholder="Email..."
  required=""
  type="email"
  value=""
/>
`;

exports[`labels render and have proper text titles 3`] = `
<textarea
  class="form-control"
  id="message"
  name="message"
  placeholder="Message..."
  required=""
/>
`;
